src/axe-raw-sarif-converter-21.test.ts

```diff
@@ -73,6 +61,42 @@ describe('AxeRawSarifConverter21', () => {
 
             expect(axeRawToSarifOutput).toEqual(axeToSarifOutput);
         });
+
+        function normalizeSarif(sarif: Sarif.Log): void {
+            sarif.runs[0].results = sortBy(sarif.runs[0].results, [
+                'ruleId',
+                'locations[0].logicalLocations[0].fullyQualifiedName',
+                'kind',
+            ]);
+
+            // remove logicalLocation that contains xpath in case xpath is not present
+            // in axe results (default xpath option = false) but present in axe raw results
+            for (let i = 0; i < sarif.runs[0].results.length; i++) {
+                if (isSecondLogicalLocationPresent(sarif.runs[0].results[i])) {
+                    sarif.runs[0].results[
+                        i
+                    ].locations![0].logicalLocations! = removeSecondLogicalLocation(
+                        sarif.runs[0].results[i].locations![0]
+                            .logicalLocations!,
+                    );
+                }
+            }
+        }
+
+        function isSecondLogicalLocationPresent(result: Sarif.Result) {
+            return (
+                result.locations &&
+                result.locations![0] &&
+                result.locations![0].logicalLocations! &&
+                result.locations![0].logicalLocations!.length > 1
+            );
+        }
+
+        function removeSecondLogicalLocation(
+            logicalLocations: Sarif.LogicalLocation[],
+        ): Sarif.LogicalLocation[] {
+            return [logicalLocations[0]];
+        }
```

Comment:
I think this is a case where it results in clearer code to write it a little less functionally:

```ts
        function removeOptionalXpathLocation(result: Sarif.Result): void {
            if (
                result.locations &&
                result.locations![0].logicalLocations!.length > 1
            ) {
                result.locations![0].logicalLocations!.pop();
            }
        }

        function normalizeSarif(sarif: Sarif.Log): void {
            // ...
            sarif.runs[0].results.forEach(removeOptionalXpathLocation);
        }
```